wildfly-elytron-credential/pom.xml

```diff
@@ -62,12 +62,36 @@
             <groupId>org.jboss.logging</groupId>
             <artifactId>jboss-logging</artifactId>
             <scope>provided</scope>
-        </dependency>        
+        </dependency>
         <dependency>
             <groupId>org.jboss.logging</groupId>
             <artifactId>jboss-logging-processor</artifactId>
             <scope>provided</scope>
-        </dependency>               
+        </dependency>
+
+        <!--Test scope-->
+        <dependency>
+            <groupId>org.wildfly.security</groupId>
+            <artifactId>wildfly-elytron-digest</artifactId>
```

Comment:
Overall everything on this PR looks really good, it looks like most of the moved tests have moved cleanly without internal dependencies - what was the need for this dependency?  It looks like this module will now depend on digest where it did not previously?

Reply:
If some tests did not move cleanly I left it in wildfly-elytron-tests. For this specific one, after addding this dependency all the tests passed. Without it the single test https://github.com/Skyllarr/wildfly-elytron/blob/tests-splitting/wildfly-elytron-credential/src/test/java/org/wildfly/security/password/impl/DigestPasswordTest.java#L94

will throw

```
testSHA512_256(org.wildfly.security.password.impl.DigestPasswordTest)  Time elapsed: 0.055 sec  <<< ERROR!
java.security.spec.InvalidKeySpecException: ELY08013: No such MessageDigest algorithm for "digest-sha-512-256"
	at org.wildfly.security.password.impl.DigestPasswordImpl.<init>(DigestPasswordImpl.java:82)
	at org.wildfly.security.password.impl.DigestPasswordImpl.<init>(DigestPasswordImpl.java:72)
	at org.wildfly.security.password.impl.PasswordFactorySpiImpl.engineGeneratePassword(PasswordFactorySpiImpl.java:261)
	at org.wildfly.security.password.PasswordFactory.generatePassword(PasswordFactory.java:151)
	at org.wildfly.security.password.impl.DigestPasswordTest.performTest(DigestPasswordTest.java:106)
	at org.wildfly.security.password.impl.DigestPasswordTest.testSHA512_256(DigestPasswordTest.java:96)
```

so I made an exception and added dependency here.